cmd/find.go

```diff
@@ -190,142 +199,129 @@ func doFind(clnt Client, ctx *cli.Context) {
 		}
 
 		fileContent := parseContent(content)
-		filePath := fileContent.Key
-
-		// traversing in a object store not a file path
-		if err == nil {
-			filePath = path.Join(alias, filePath)
+		if targetAlias != "" {
+			fileContent.Key = path.Join(targetAlias, fileContent.Key)
 		}
 
 		if ctx.String("maxdepth") != "" {
-			i, e := strconv.Atoi(ctx.String("maxdepth"))
-			s := ""
-
+			maxDepth, e := strconv.Atoi(ctx.String("maxdepth"))
 			fatalIf(probe.NewError(e), "Error parsing string passed to flag maxdepth")
 
-			// we are going to be parsing the path by x amounts
-			pathParts := strings.SplitAfter(filePath, "/")
+			newPath := ""
 
-			// handle invalid params
-			// ex. user specifies:
-			// maxdepth 2, but the given object only has a maxdepth of 1
-			if (len(pathParts)-1) < i || i < 0 {
+			// We are going to be parsing the path by x amounts
+			pathParts := strings.SplitAfter(fileContent.Key, separator)
 
-				// -1 is meant to handle each the array being 0 indexed, but the size not being 0 indexed
-				i = len(pathParts) - 1
+			// Check if max-depth is 2, but the given object only
+			// has a maximum depth of 1 use that instead.
+			if (len(pathParts)-1) < maxDepth || maxDepth < 0 {
+				// -1 is meant to handle each the array being 0 indexed,
+				// but the size not being 0 indexed
+				maxDepth = len(pathParts) - 1
 			}
 
-			// append portions of path into a string
-			for j := 0; j <= i; j++ {
-				s += pathParts[j]
+			// Construct new path based on the requested maxDepth.
+			for j := 0; j <= maxDepth; j++ {
+				newPath += pathParts[j]
 			}
 
-			filePath = s
-			fileContent.Key = s
+			fileContent.Key = newPath
 		}
 
-		// maxdepth can modify the filepath to end in a directory
-		// to be consistent with find we do not want to be listing directories
-		// so any parms which end in / will be ignored
-		if !strings.HasSuffix(filePath, "/") {
-
-			orBool := ctx.Bool("or")
-
-			match := fileContentMatch(fileContent, orBool, ctx)
-
-			if match && ctx.String("print") != "" {
-				doFindPrint(filePath, ctx, fileContent)
-			} else if match {
-				printMsg(findMSG{fileContent})
-			}
-
-			if !ctx.Bool("watch") && match && ctx.String("exec") != "" {
-				doFindExec(ctx, filePath, fileContent)
-			}
+		// Maxdepth can modify the filepath to end as a directory prefix
+		// to be consistent with the find behavior, we wont list directories
+		// so any paths which end with a separator are ignored.
+		if strings.HasSuffix(fileContent.Key, separator) {
+			continue
 		}
 
+		match := doFindMatch(ctx, fileContent)
+		if !ctx.Bool("watch") && match && ctx.String("exec") != "" {
+			doFindExec(ctx, fileContent)
+		} else if match && ctx.String("print") != "" {
+			doFindPrint(ctx, fileContent)
+		} else if match {
+			printMsg(findMessage{fileContent})
+		}
 		if ctx.Bool("watch") {
-			doFindWatch(ctx.Args().Get(0), ctx)
+			doFindWatch(ctx, ctx.Args().Get(0))
```

Comment:
can I pass watch and exec as part of the same command as given in the example 7?
```
 mc find s3/buck --name "*foo" --watch --exec "mc cp {} play/test"

```
If I can, does doFindWatch do both watch and exec internally?

Reply:
Yes that is intended.... watch does exec internally.